### PR TITLE
[PHPSTAN] Fill empty php doc param types (DataObject)

### PR DESCRIPTION
**Pull request by blankse**
Fix many (not all) phpstan level 2 errors in Pimcore\Model\DataObject namespace with following schema:
```
PHPDoc tag @param has invalid value ($xyz)
```
Before:
```
 [ERROR] Found 2731 errors
```
After:
```
 [ERROR] Found 2267 errors
```

### CODE REVIEW

File: models/DataObject/ClassDefinition/CustomLayout/Dao.php [hunk 1]
```diff
@@ -107,7 +107,7 @@ public function getIdByNameAndClassId($name, $classId)
     }
 
     /**
-     * @return int|mixed
+     * @return int
```
Comment (brusch) [inline models/DataObject/ClassDefinition/CustomLayout/Dao.php]: Should be `int|null` right?

File: models/DataObject/Data/ObjectMetadata.php [hunk 2]
```diff
@@ -117,23 +117,23 @@ public function __call($name, $arguments)
     /**
      * @param $object
      * @param string $ownertype
-     * @param $ownername
-     * @param $position
-     * @param $index
+     * @param string $ownername
+     * @param int $position
```
Comment (brusch) [inline models/DataObject/Data/ObjectMetadata.php]: `$position`  can be a string as well (e.g. localized fields).

File: models/DataObject/Data/ObjectMetadata.php [hunk 3]
```diff
@@ -117,23 +117,23 @@ public function __call($name, $arguments)
...
-     * @param $fieldname
-     * @param $ownertype
-     * @param $ownername
-     * @param $position
-     * @param $index
+     * @param int $destinationId
+     * @param string $fieldname
+     * @param string $ownertype
+     * @param string $ownername
+     * @param int $position
```
Comment (brusch) [inline models/DataObject/Data/ObjectMetadata.php]: Same here: `string`

File: models/DataObject/ClassDefinition/CustomLayout/Dao.php (hunk 1 above)
Comment (blankse) [reply models/DataObject/ClassDefinition/CustomLayout/Dao.php]: Why? If the DB return null, the value will be set to 1

Comment (brusch) [reply models/DataObject/ClassDefinition/CustomLayout/Dao.php]: You're right, sorry!

File: models/DataObject/Data/ObjectMetadata.php (hunk 2 above)
Comment (blankse) [reply models/DataObject/Data/ObjectMetadata.php]: You are right. I changed it

File: models/DataObject/Data/ObjectMetadata.php (hunk 3 above)
Comment (blankse) [reply models/DataObject/Data/ObjectMetadata.php]: You are right. I changed it

**Pull request merged**